storage: enabling s3 storage support.

matrixji opened the pull request:
- very raw idea for support s3 storage, not enabled by default.
- need enabling compiled(cmake) with MILVUS_WITH_AWS.
- simple integration tested with ceph/radowsgw.

Signed-off-by: Ji Bin <matrixji@live.com>

**What type of PR is this?**

- [ ] API-change
- [ ] BUG
- [x] Improvement
- [ ] Documentation
- [ ] Feature
- [ ] Test and CI
- [ ] Code Refactoring

**Which issue(s) this PR fixes:**

Fixes #1434 

**What this PR does / why we need it:**

Enabling S3 support on 1.x, currently I'm trying to running milvus with some s3 storage system, e.g Ceph's radowsgw.

**Special notes for your reviewer:**

Not Available

**Additional documentation (e.g. design docs, usage docs, etc.):**

Not Available

matrixji committed:
storage: enabling s3 storage support.

- very raw idea for support s3 storage, not enabled by default.
- need enabling compiled(cmake) with MILVUS_WITH_AWS.
- simple integration tested with ceph/radowsgw.

Signed-off-by: Ji Bin <matrixji@live.com>

yhmo commented on core/src/codecs/default/DefaultIdBloomFilterFormat.cpp:
```diff
@@ -36,8 +37,11 @@ DefaultIdBloomFilterFormat::read(const storage::FSHandlerPtr& fs_ptr, segment::I
+    scaling_bloom_t* bloom_filter{nullptr};
+    if (fs_ptr->operation_ptr_->CacheGet(bloom_filter_file_path)) {
+        bloom_filter =
+            new_scaling_bloom_from_file(bloom_filter_capacity, bloom_filter_error_rate, bloom_filter_file_path.c_str());
```
The bloom_filter internally use mmap() to mapping a file handler. This could be a risk point.

matrixji replied:
IMO, For reading bloom filter, it's OK, since the file if already exist, the segment writer should take responsible for it's correctness, but in DefaultIdBloomFilterFormat::create, CachePut may not valid, since the file content is not synced.

yhmo replied:
You are right. 
In fact, in 0.11.0(experimental), we rewrite the bloom filter code, remove the mmap(). After the change, the read()/write() logic become more clear:
https://github.com/milvus-io/milvus/blob/experimental/core/src/codecs/IdBloomFilterFormat.cpp

To avoid this risk, we need to do same change in 1.x. But it is a next step work, not relate with this PR.

matrixji committed:
storage: enabling s3, rebase and improvments

- Do not do CachePut in DefaultIdBloomFilterFormat::create, since bloom
fileter file may not synced with memory at that moment.

Signed-off-by: Ji Bin <matrixji@live.com>
